File: code/jvm/im-http-api/src/main/kotlin/im/api/model/output/messages/MessageOutputModel.kt
package im.api.model.output.messages

import im.api.model.output.users.UserOutputModel
import im.domain.messages.Message
import java.time.LocalDateTime

data class MessageOutputModel(
    val id: Long,
    val channelId: Long,
    val author: UserOutputModel,
    val content: String,
    val createdAt: LocalDateTime,
    val editedAt: LocalDateTime?,
) {
    companion object {
        fun fromDomain(message: Message): MessageOutputModel =
            MessageOutputModel(
                id = message.id.value,
                channelId = message.channelId.value,
                author = UserOutputModel.fromDomain(message.user),
                content = message.content,
                createdAt = message.createdAt,
                editedAt = message.editedAt,
            )
    }
}

File: code/client/web/src/Dto/output/messages/MessageOutputModel.ts
import { UserOutputModel } from '../users/UserOutputModel';

/**
 * Output model for message
 *
 * @param id The id of the message.
 * @param channelId The id of the channel the message belongs to.
 * @param author The author of the message.
 * @param content The content of the message.
 * @param createdAt The date and time when the message was created.
 * @param editedAt The date and time when the message was last edited.
 */
export interface MessageOutputModel {
    id: number;
    channelId: number;
    author: UserOutputModel;
    content: string;
    createdAt: Date;
    editedAt: Date | null;
}
